include/SDL3/SDL_video.h

```diff
@@ -743,6 +743,9 @@ extern DECLSPEC const char *SDLCALL SDL_GetWindowTitle(SDL_Window *window);
 /**
  * Set the icon for a window.
  *
+ * Systems that do not support setting an icon to the window, will always fail,
+ * reporting `SDL_UNSUPPORTED`.
```

Comment:
I don't think that extra complexity of returning `SDL_UNSUPPORTED` is necessary.

Reply:
I guess this doc addition can be dropped now?